lib/saluki-components/src/destinations/datadog_metrics/mod.rs

```diff
@@ -552,14 +569,21 @@ async fn run_endpoint_io_loop<S, B>(
     task_barrier.wait().await;
 }
 
-fn for_resolved_endpoint<B>(endpoint: ResolvedEndpoint) -> impl Fn(Request<B>) -> Request<B> + Clone {
+#[allow(unused)]
+fn for_resolved_endpoint<B>(
+    endpoint: ResolvedEndpoint, refresher: Arc<RefreshableConfiguration>,
+) -> impl Fn(Request<B>) -> Request<B> + Clone {
     let new_uri_authority = Authority::try_from(endpoint.endpoint().authority())
         .expect("should not fail to construct new endpoint authority");
     let new_uri_scheme =
         Scheme::try_from(endpoint.endpoint().scheme()).expect("should not fail to construct new endpoint scheme");
-    let api_key_value =
-        HeaderValue::from_str(endpoint.api_key()).expect("should not fail to construct API key header value");
-
+    let mut api_key: String;
+    if let Ok(refresher_api_key) = refresher.get_typed::<String>("api_key") {
+        api_key = refresher_api_key.clone();
+    } else {
+        api_key = endpoint.api_key().to_string();
+    }
```

Comment:
This is going to apply the API key from the refreshable configuration to _all_ resolved endpoints, which is not what we want to do.

We might need to consider a design where we shove `RefreshableConfiguration` into `ResolvedEndpoint` and move some stuff around, so that we can amortize the cost using the refresher and only apply it where desired.